Support stacked bar series labels

alxnddr opened the pull request:
Closes https://github.com/metabase/metabase/issues/41980  https://github.com/metabase/metabase/issues/13096

### Description

Adds the ability to show data labels of individual series on stacked bar charts.

### How to verify

- create a stacked bar/combo charts
- enable showing individual series labels using "Stack values to show"

### Demo

https://github.com/metabase/metabase/assets/14301985/766a37ef-ef1e-4c9e-bc5e-c853ec3d4c5c

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

EmmadUsmani commented on frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts:
```diff
@@ -82,16 +84,44 @@ export const getBarLabelLayout =
+      };
+    }
+
+    const labelValue = dataset[dataIndex][seriesDataKey];
```
Should use `transformedDataset[ORIGINAL_INDEX_DATA_KEY]` for consistency with `buildEChartsStackLabelOptions`, or in `buildEChartsStackLabelOptions` just use `params.dataIndex` for consistency with this function

JesseSDevaney commented on frontend/src/metabase/visualizations/lib/settings/graph.unit.spec.js:
```diff
@@ -286,4 +281,79 @@ describe("GRAPH_DISPLAY_VALUES_SETTINGS", () => {
+      expect(isHidden).toBe(false);
+    });
+
+    it("should be hidden on normalized charts bar charts", () => {
+      const isHidden = getHidden(
+        [{ card: { display: "bar" } }, { card: { display: "bar" } }],
+        {
+          series: series => ({ display: series.card.display }),
+          "stackable.stack_type": "normalized",
+          "graph.show_values": true,
+        },
+      );
+
+      expect(isHidden).toBe(true);
+    });
```
If this is hidden on normalized bar charts (which I believe makes sense), we need to act as if the value is set to `"total"` because otherwise the `Show values on data points` setting does not make sense. It does not allow the user to configure whether they want inside labels or not in `Stack - 100%`.

But also considering the case where we have a normalized bar chart with a regular line series added, we should still allow the inside labels to be configurable. Like still show the setting but only show valid options for it like `None` or `Insides`.

![image](https://github.com/metabase/metabase/assets/22608765/804bdf6a-9d04-486e-8eee-ea4e98e16fe0)

JesseSDevaney replied:
Also I know the product doc displays the labels with the absolute values, but do we want to instead to display the relative percentages in a normalized bar chart (`Stack - 100%`). To me this makes more sense to show, because the user intentionally selected normalized, but we are still showing absolute values?

![image](https://github.com/metabase/metabase/assets/22608765/0f13aade-71b2-4845-bff6-4e2d41b5407e)

JesseSDevaney reviewed (changes requested):
Hi Sasha! Nice work on this overall :heart: 

This feels 99% of the way there, but there are still a couple things I think we should address before merging.

---
1. We should probably hide the `Values to show` setting in `Stack - 100%` WITHOUT a line series because it will not affect the individual labels for stacked series. Or does it? I could not find a space where it was affecting whether some inside labels were displayed or hidden.
![image](https://github.com/metabase/metabase/assets/22608765/5a031ef7-9a91-4f75-bdd1-f81d04a2f734)
2. Also, we no longer display the `Auto formatting` setting which still applies to these labels. See above and below for comparison. The setting is carried over but not adjustable in `Stack - 100%` mode. 
![image](https://github.com/metabase/metabase/assets/22608765/724f24e8-9ab7-4ec6-901d-b0cd1f21dcfe)
_(assuming if we still want to display absolute values instead of relative percentages - relative percentages makes more sense to me considering the specified chart type)_
3. (in auto formatting mode) if one series is compact, we should probably compact the labels for the rest of the series. a mix of compact and full feels wonky.
![image](https://github.com/metabase/metabase/assets/22608765/44a436e3-e99a-4870-a76e-011e1fde01ab)
4. I also left some individual comments on other areas in the code.

alxnddr commented:
@JesseSDevaney thanks for the review, I addressed your feedback here, please have a look https://github.com/metabase/metabase/pull/42803/commits/8ad8bed92ccf45450b1c7daeb299e9140c0a2c20

alxnddr commented on .loki/reference/chrome_laptop_static_viz_ComboChart_Bar_Stacked_Normalized_Series_Labels.png:
1. I picked the `power` scale to have various sizes bars. There should not be any harm with this so I think we can keep it.
2. Having non-linear scales on stacked charts is a questionable request, however, since we supported them with dc.js we reproduced it with ECharts as well in case if some customers use them

JesseSDevaney reviewed (changes requested):
Sasha, thank you for addressing my comments! :heart: 

There are a couple small things left to polish and then we should be good to go

alxnddr committed:
support stacked series values labels

alxnddr committed:
Merge branch 'master' into support-stacked-series-labels
